src/Service/Program.cs

```diff
@@ -62,46 +65,46 @@ public static bool StartEngine(string[] args)
                 });
 
         /// <summary>
-        /// Iterate through args and based on values present
-        /// set the appropriate log level. If --LogLevel is present
-        /// the next value in args must be "0" through "6", or
-        /// --LogLevel must be the last element in args. In any other
-        /// case we throw an exception. If --LogLevel is the last element
-        /// in args then we ignore it to maintain engine's behavior prior
-        /// to this change.
+        /// Using System.CommandLine Parser to parse args and return
+        /// the correct log level. We save if there is a log level in args through
+        /// the out param. For log level out of range we throw an exception.
         /// </summary>
         /// <param name="args">array that may contain log level information.</param>
         /// <param name="isLogLevelOverridenByCli">sets if log level is found in the args.</param>
         /// <returns>Appropriate log level.</returns>
         private static LogLevel GetLogLevelFromCommandLineArgs(string[] args, out bool isLogLevelOverridenByCli)
         {
-            for (int i = 0; i < args.Length; i++)
-            {
-                if (args[i].Equals("--LogLevel"))
-                {
-                    if (args.Length <= i + 1)
-                    {
-                        break;
-                    }
+            Command cmd = new(name: "start");
+            Option<LogLevel> logLevelOption = new(name: "--LogLevel");
+            cmd.AddOption(logLevelOption);
+            ParseResult result = GetParseResult(cmd, args);
+            LogLevel logLevel = result.Tokens.Count > 1 ? result.GetValueForOption<LogLevel>(logLevelOption) : LogLevel.Error;
+            isLogLevelOverridenByCli = result.Tokens.Count > 1 ? true : false;
 
-                    if (Enum.TryParse(args[i + 1], out LogLevel logLevel))
-                    {
-                        isLogLevelOverridenByCli = true;
-                        return logLevel;
-                    }
-                    else
-                    {
-                        throw new DataApiBuilderException(
-                            message: $"LogLevel's valid range is 0 to 6, your value: {args[i]}, see: " +
-                            $"https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loglevel?view=dotnet-plat-ext-7.0",
-                            statusCode: System.Net.HttpStatusCode.ServiceUnavailable,
-                            subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
-                    }
-                }
+            if (logLevel is > LogLevel.None or < LogLevel.Trace)
```

Comment:
```suggestion
            if (logLevel > LogLevel.None || logLevel < LogLevel.Trace)
```

can you point to reference for the syntax you used if that was building without error?

Reply:
using the `is operator` is allowed for property patterns from what I understand.

https://learn.microsoft.com/en-US/dotnet/csharp/language-reference/operators/patterns

Reply:
No need to rewrite, thanks for the link. I did not recognize that pattern matching applied to if conditionals. I had only seen it used for switch.